test/FsAutoComplete.Tests.Lsp/Program.fs

```diff
@@ -63,63 +69,65 @@ let lspTests =
     "lsp"
     [ for (loaderName, workspaceLoaderFactory) in loaders do
         for (lspName, lspFactory) in lspServers do
-          testList
-            $"{loaderName}.{lspName}"
-            [ 
-              Templates.tests ()
-              let createServer () =
-                lspFactory toolsPath workspaceLoaderFactory
-
-              initTests createServer
-              closeTests createServer
-
-              Utils.Tests.Server.tests createServer
-              Utils.Tests.CursorbasedTests.tests createServer
-
-              CodeLens.tests createServer
-              documentSymbolTest createServer
-              Completion.autocompleteTest createServer
-              Completion.autoOpenTests createServer
-              foldingTests createServer
-              tooltipTests createServer
-              Highlighting.tests createServer
-              scriptPreviewTests createServer
-              scriptEvictionTests createServer
-              scriptProjectOptionsCacheTests createServer
-              dependencyManagerTests createServer
-              interactiveDirectivesUnitTests
-
-              // commented out because FSDN is down
-              //fsdnTest createServer
-
-              //linterTests createServer
-              uriTests
-              formattingTests createServer
-              analyzerTests createServer
-              signatureTests createServer
-              SignatureHelp.tests createServer
-              CodeFixTests.Tests.tests createServer
-              Completion.tests createServer
-              GoTo.tests createServer
-
-              FindReferences.tests createServer
-              Rename.tests createServer
-
-              InfoPanelTests.docFormattingTest createServer
-              DetectUnitTests.tests createServer
-              XmlDocumentationGeneration.tests createServer
-              InlayHintTests.tests createServer
-              DependentFileChecking.tests createServer
-              UnusedDeclarationsTests.tests createServer
-
-              ] ]
-      
+          for (sourceTextName, sourceTextFactory) in sourceTextFactories do
```

Comment:
more pivots :D 

One thing I was thinking about was for perf reasons we may want to start splitting these jobs out into separate CI runs, where each permutation runs as a distinct job.